server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterInfoSimulatorTests.java

```diff
@@ -179,41 +279,238 @@ public void testRelocateShardWithMultipleDataPath1() {
         );
     }
 
+    public void testInitializeShardFromSnapshot() {
+
+        var shardSize = 100;
+        var indexSettings = indexSettings(IndexVersion.current(), 1, 0);
+        if (randomBoolean()) {
+            indexSettings.put(INDEX_STORE_TYPE_SETTING.getKey(), SEARCHABLE_SNAPSHOT_STORE_TYPE);
+        }
+
+        var state = ClusterState.builder(ClusterName.DEFAULT)
+            .metadata(Metadata.builder().put(IndexMetadata.builder("my-index").settings(indexSettings)))
+            .build();
+
+        var snapshot = new Snapshot("repository", new SnapshotId("snapshot-1", "na"));
+        var indexId = new IndexId("my-index", "_na_");
+        var shard = newShardRouting(
+            new ShardId(state.metadata().index("my-index").getIndex(), 0),
+            "node-0",
+            true,
+            ShardRoutingState.INITIALIZING,
+            new RecoverySource.SnapshotRecoverySource(randomUUID(), snapshot, IndexVersion.current(), indexId)
+        );
+
+        var initialClusterInfo = new ClusterInfoTestBuilder() //
+            .withNode("node-0", new DiskUsageBuilder(1000, 1000))
+            .withNode("node-1", new DiskUsageBuilder(1000, 1000))
+            .build();
+        var snapshotShardSizeInfo = new SnapshotShardSizeInfoTestBuilder() //
+            .withShard(snapshot, indexId, shard.shardId(), shardSize)
+            .build();
+
+        var allocation = createRoutingAllocation(state, initialClusterInfo, snapshotShardSizeInfo);
+        var simulator = new ClusterInfoSimulator(allocation);
+        simulator.simulateShardStarted(shard);
+
+        assertThat(
+            simulator.getClusterInfo(),
+            equalTo(
+                new ClusterInfoTestBuilder() //
+                    .withNode("node-0", new DiskUsageBuilder(1000, 1000 - shardSize))
+                    .withNode("node-1", new DiskUsageBuilder(1000, 1000))
+                    .withShard(shard, shardSize)
+                    .build()
+            )
+        );
+    }
+
+    public void testInitializeShardFromSearchableSnapshot() {
```

Comment:
nit

```suggestion
    public void testInitializeShardFromPartialSearchableSnapshot() {
```

since the simple (non-partial) searchable snapshot case is tested in the test above with a randomBool.